### ISSUE TITLE
Cannot send numbers or bools as payloads

### ISSUE THREAD

**Issue by DonutEspresso** [restify/node-restify#1607]
- [x] Used appropriate template for the issue type
- [x] Searched both open and closed issues for duplicates of this issue
- [x] Title adequately and _concisely_ reflects the feature or the bug

# Bug Report

A regression of GH-800. You cannot send numbers or false even with an explicit status code. i.e.,

```
res.send(200, false);
res.send(200, true);
res.send(200, 0);
```

These all cause restify to blow up. 

## Restify Version

6.x

## Node.js Version

All

## Cause

Looks like this the arguments parsing in `lib/response.js`'s `__send` method:

```
        if (
            typeof arguments[index] === 'object' ||
            typeof arguments[index] === 'string'
        ) {
            body = arguments[index++];
        }
```

The type checks explicitly disallow numbers and bools. 

## Are you willing and able to fix this?

Yes

**Comment by retrohacker** [comment]
This is a duplicate of https://github.com/restify/node-restify/issues/1461 I believe

**Comment by retrohacker** [comment]
I think we settled on fixing this, with the following caveat:

> number is a special case in that there is a previous argument that is a number: `statusCode`. I'd err on the side of keeping the order of the args, in the case where you want to specify a number for `body`, you must always specify `statusCode` to disambiguate.

* `res.send(false);`
* `res.send(200, false);`
* `res.send(100);` // status code of 100
* `res.send(200, 100);` // status code of 200, body of 100

**Comment by retrohacker** [comment]
So essentially `statusCode` is optional for everything except a number. If you provide a number as a first argument, it will _always_ be a `statusCode`.

**Comment by DonutEspresso** [comment]
Yep, makes sense, I think we're just missing #2 and #4 from your list.

**Comment by DonutEspresso** [comment]
Fixed by #1609
